eng/Subsets.props

```diff
@@ -203,7 +203,7 @@
 
   <!-- Installer sets -->
   <ItemGroup Condition="$(_subset.Contains('+corehost+'))">
-    <CorehostProjectToBuild Include="$(InstallerProjectRoot)corehost\build.proj" SignPhase="Binaries" />
+    <CorehostProjectToBuild Include="$(InstallerProjectRoot)corehost\corehost.proj" SignPhase="Binaries" />
```

Comment:
The reason why this project needs to be renamed is because we discovered that the project.assets.json file for this installer build.proj clashed with the repo's root build.proj, causing restore issues and overrides whenever the installer subset was being built. This will fix that issue. Kudos for finding the issue to @safern and @ViktorHofer